src/pharmpy/tools/mfl/parse.py

```diff
@@ -309,6 +315,38 @@ def replace(self, **kwargs):
             metabolite=metabolite,
         )
 
+    def replace_features(self, mfl_str):
+        key_dict = {
+            Absorption: 'absorption',
+            Elimination: 'elimination',
+            Transits: 'transits',
+            Peripherals: 'peripherals',
+            LagTime: 'lagtime',
+            Covariate: 'covariate',
+            DirectEffect: 'direct_effect',
+            EffectComp: 'effect_comp',
+            IndirectEffect: 'indirect_effect',
+            Metabolite: 'metabolite',
+        }
+        mfl_list = _parse(mfl_str)
+        kwargs = {}
+        for statement in mfl_list:
+            key = key_dict[type(statement)]
+            if type(statement) in (Transits, Peripherals, Covariate, IndirectEffect):
+                no_of_modes = len(statement.counts)
+                value = (statement,)
+            else:
+                no_of_modes = len(statement.modes)
+                value = statement
+            if no_of_modes > 1:
+                raise ValueError(f'Cannot set more than one mode for type: `{key}`')
+            if key_dict[Peripherals] in kwargs.keys():
+                kwargs[key] += value
+            else:
+                kwargs[key] = value
```

Comment:
As for instance `Covariate` effects can be given in more than one statement, wouldn't this overwrite the previously stored value if multiple statements are given?
This goes hand in hand with my question above regarding if this should be supported

Reply:
`Covariate` should be included, an oversight. Will change!